src/firefly/js/ui/panel/CollapsiblePanel.jsx

```diff
@@ -25,85 +25,53 @@ export const CollapseHeaderCorner = {
                             BottomLeft: 0b1000
                         };
 
-function collapsibleStateFromProps(props) {
-    // component key needs to be defined if the state needs to be saved though unmount/mount
-    let isOpen = props.componentKey && getComponentState(props.componentKey).isOpen;
-    if (!isBoolean(isOpen)) {
-        // use property to initialize state
-        isOpen = props.isOpen? props.isOpen: false;
-    }
 
-    return {isOpen};
+export function CollapsiblePanel(props){
+    const isOpenState = useStoreConnector(()=>(getComponentState(props.componentKey)?.isOpen ?? props.isOpen));
+    const newProps = {...props,
+            isOpen : isOpenState,
+            onToggle: (newIsOpen)=>{
+                dispatchComponentStateChange(props.componentKey, {isOpen: newIsOpen});
+                props.onToggle?.(newIsOpen);
+        }};
+
+    return <CollapsiblePanelView {...newProps} />;
 }
 
-export class CollapsiblePanel extends PureComponent {
 
-    constructor(props) {
-        super(props);
 
-        this.state = collapsibleStateFromProps(props);
+export function CollapsiblePanelView({header, isOpen=false, headerRoundCorner=0, headerStyle, contentStyle,
+                                     style, borderStyle=CollapseBorder.Oneborder, onToggle, children }) {
+    const contentBorderClassName = ['', ' CollapsiblePanel__Content-oneborder',
+        ' CollapsiblePanel__Content-twoborder', ' CollapsiblePanel__Content-threeborder'];
+    let headerClassName = 'CollapsiblePanel__Header';
+    let contentClassName = 'CollapsiblePanel__Content';
+    const headerCorner = Object.values(CollapseHeaderCorner).reduce(
+        (headerCornerStr, corner) => headerCornerStr + (corner&headerRoundCorner ? ' 0.5em' : ' 0em'), '');
```

Comment:
Nice- better than my suggestion!

Reply:
I scanned through this PR and was puzzled at this block of code.  It has nothing to do with Jaladh changes.  I am referring to 
```
export const CollapseBorder = {
                            Noborder: 0,
                            Oneborder: 1,    // left side
                            Twoborder: 2,    // left and right sides
                            Threeborder: 3   // left, bottom and right sides
                        };
export const CollapseHeaderCorner = {
                            TopLeft: 0b0001,
                            TopRight: 0b0010,
                            BottomRight: 0b0100,
                            BottomLeft: 0b1000
                        };
```
It's unnecessarily complex.  And imo, not how you style a component.  `borderRadius` is now fixed(cannot override) to either 0 or 0.5em.  The good part is, it's not used anywhere.  I suggest we remove this complex logic and add the `Oneborder` caes into `CollapsiblePanel__Header` as default.  Of course, one can override it by setting `headerStyle` already.